commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/CommonsLangPortedFractionTest.java

```diff
@@ -0,0 +1,1110 @@
+        f = f.abs();
+        assertEquals(2, f.getNumerator());
+        assertEquals(3, f.getDenominator());
+
+        f = Fraction.of(-50, 75);
+        f = f.abs();
+        assertEquals(2, f.getNumerator());
+        assertEquals(3, f.getDenominator());
+
+        f = Fraction.of(Integer.MAX_VALUE, 1);
+        f = f.abs();
+        assertEquals(Integer.MAX_VALUE, f.getNumerator());
+        assertEquals(1, f.getDenominator());
+
+        f = Fraction.of(Integer.MAX_VALUE, -1);
+        f = f.abs();
+        assertEquals(-Integer.MAX_VALUE, f.getNumerator());
+        assertEquals(-1, f.getDenominator());
+
+        Fraction.of(Integer.MIN_VALUE, 1).abs();
+    }
+
+    @Test
+    public void testPow() {
+        Fraction f = null;
+
+        f = Fraction.of(3, 5);
+        assertEquals(Fraction.ONE, f.pow(0));
+
+        f = Fraction.of(3, 5);
+        assertEquals(f, f.pow(1));
+        assertEquals(f, f.pow(1));
+
+        f = Fraction.of(3, 5);
+        f = f.pow(2);
+        assertEquals(9, f.getNumerator());
+        assertEquals(25, f.getDenominator());
+
+        f = Fraction.of(3, 5);
+        f = f.pow(3);
+        assertEquals(27, f.getNumerator());
+        assertEquals(125, f.getDenominator());
+
+        f = Fraction.of(3, 5);
+        f = f.pow(-1);
+        assertEquals(5, f.getNumerator());
+        assertEquals(3, f.getDenominator());
+
+        f = Fraction.of(3, 5);
+        f = f.pow(-2);
+        assertEquals(25, f.getNumerator());
+        assertEquals(9, f.getDenominator());
+
+        // check unreduced fractions stay that way.
+        f = Fraction.of(6, 10);
+        assertEquals(Fraction.ONE, f.pow(0));
+
+        f = Fraction.of(6, 10);
+        assertEquals(f, f.pow(1));
+        assertEquals(f.pow(1), Fraction.of(3, 5));
+
+        f = Fraction.of(6, 10);
+        f = f.pow(2);
+        assertEquals(9, f.getNumerator());
+        assertEquals(25, f.getDenominator());
+
+        f = Fraction.of(6, 10);
+        f = f.pow(3);
+        assertEquals(27, f.getNumerator());
+        assertEquals(125, f.getDenominator());
+
+        f = Fraction.of(6, 10);
+        f = f.pow(-1);
+        assertEquals(5, f.getNumerator());
+        assertEquals(3, f.getDenominator());
+
+        f = Fraction.of(6, 10);
+        f = f.pow(-2);
+        assertEquals(25, f.getNumerator());
+        assertEquals(9, f.getDenominator());
+
+        // zero to any positive power is still zero.
+        f = Fraction.of(0, 1231);
+        f = f.pow(1);
+        assertEquals(0, f.compareTo(Fraction.ZERO));
+        assertEquals(0, f.getNumerator());
+        assertEquals(1, f.getDenominator());
+        f = f.pow(2);
+        assertEquals(0, f.compareTo(Fraction.ZERO));
+        assertEquals(0, f.getNumerator());
+        assertEquals(1, f.getDenominator());
+
+        // zero to negative powers should throw an exception
+        final Fraction fr = f;
+        assertThrows(ArithmeticException.class, () -> fr.pow(-1));
```

Comment:
This is currently failing. By design Fraction 0 / 1 is zero when raised to any non-zero power and 1 / 1 when raised to the power 0.

Technically zero to the power of 0 is undefined (i.e. throw an exception). But in the interest of usability it can be represented as 1 so that computations can be performed for all numbers.

There should be tests for this in the current test suite. Perhaps this is best addressed by more comments in the Fraction class on expected behaviour.

Reply:
@aherbert So, we should change this test to non-throw, and do not throw in this situation in Fractions and BigFractions, and add some comments for this "feature". If I understand correctly?

Reply:
Wait until the fix to the pow() function in numbers-150. I think this test here is correct and numbers should change to throw the exception when zero is raised to a negative power.

Reply:
@aherbert OK, I understand now. thanks.